lib/commands/connect/shared.js

```diff
@@ -109,77 +110,69 @@ let request = exports.request = function(token, method, path, data){
   return promise;
 };
 
-let withUserConnections = exports.withUserConnections = function(token, appName, flags, allowNone, heroku){
+let withUserConnections = exports.withUserConnections = co.wrap(function* (token, appName, flags, allowNone, heroku){
   global_config();
 
   let resource_id = null;
   if (flags && flags.resource) {
     console.log("Calling Platform API for resouce id");
-    return Q.Promise(function(resolve, reject) {
-      return heroku.apps(appName).addons(flags.resource).info(function(err, addon) {
-        if(err) {
-          if (err.statusCode === 404) {
-            console.log("Connection with resource name '" + flags.resource + "' not found");
-            reject();
-          } else {
-            console.log(err);
-            reject(err);
-          }
+
+    return co(function* () {
+      try {
+        let addon = yield heroku.apps(appName).addons(flags.resource).info();
+      } catch(err) {
+        if (err.statusCode === 404) {
+          return yield Promise.reject("Connection with resource name '" + flags.resource + "' not found");
```

Comment:
I honestly don't know.  It depends on how the co package is going to handle it.  I suspect it might just be `throw err` since this generator is wrapped in `co` which returns a Promise and all exceptions get handled by the Promise`s `.catch` handler.  I would want to experiment with it to be confident in what's correct.  What you have might just be fine.